distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Encoding.enso

```diff
@@ -48,19 +49,49 @@ type Encoding
             charset = Charset.forName name
             Encoding.Value charset.name
 
+    ## ICON convert
+       A default encoding that will try to guess the encoding based on some
+       heuristics.
+
+       If a BOM is present, it will be used to determine UTF-8 or UTF-16
+       encoding. Otherwise, the input is decoded using UTF-8 unless it contains
+       invalid UTF-8 sequences, in which case Windows-1252 is used as a fallback.
+
+       When used for encoding, it will always encode using UTF-8.
+       This encoding cannot be passed to some functions that require a Java
+       Charset.
+    Default
+
     ## PRIVATE
-       Create a new Encoding object.
+       An encoding identified by its Java name.
 
        Arguments:
        - character_set: java.nio.charset name.
-    Value (character_set:Text)
+    Value (internal_character_set:Text)
```

Comment:
This cannot be done easily because we cannot have both private and public constructors mixed, and `Encoding.Default` is public.

We could make it private and create a 'factory' method `Encoding.default` I guess. Will amend in the next PR.

Reply:
Amended in this PR after all, for clarity.